### PR TITLE
Enable using client certificate authentication

### PR DESCRIPTION
**Pull request by lumberj**

### CODE REVIEW

**Comment by lumberj** [comment]
cc: @geertweening @justmoon @gip

**Comment by sentientwaffle** [comment]
LGTM

File: src/lib/config.js [hunk 1]
```diff
@@ -11,13 +13,65 @@ function isRunningTests () {
...
 }
 
+function parseCredentialsEnv () {
+  return JSON.parse(Config.getEnv(envPrefix, 'CREDENTIALS') || '{}')
+}
+
+function parseCredentials () {
+  const credentialsEnv = parseCredentialsEnv()
+
+  return _.reduce(credentialsEnv, (parsed, credentials, ledger) => {
```
Comment (clark800) [inline src/lib/config.js]: `keyBy` and `map` might be simpler than `reduce`

File: src/lib/config.js [hunk 2]
```diff
@@ -11,13 +13,65 @@ function isRunningTests () {
...
+
+    return parsed
+  }, {})
+}
+
+function validateLocalEnvConfig () {
+  const credentials = parseCredentialsEnv()
+
+  _.forEach(credentials, (credential, ledger) => {
+    if (credential.username === undefined ^ credential.password === undefined) {
```
Comment (clark800) [inline src/lib/config.js]: This does work, but `^` is the bitwise XOR operator, not a boolean XOR operator, which means this returns `0` or `1`, which would be a type error if we apply type checking

Comment (clark800) [inline src/lib/config.js]: Note: boolean XOR operator is `!==`

File: test/mocha.opts [hunk 3]
```diff
@@ -0,0 +1 @@
+-r co-mocha
```
Comment (clark800) [inline test/mocha.opts]: With this I think we can remove the option from package.json

File: src/lib/config.js (hunk 2 above)
Comment (lumberj) [inline src/lib/config.js]: Sure, I can change this to 

```js
credential.username === undefined !== credential.password === undefined
```

Comment (clark800) [inline src/lib/config.js]: don't forget parens :)

File: src/lib/config.js (hunk 1 above)
Comment (lumberj) [inline src/lib/config.js]: I can refactor this in the future, as it would require a major version bump for lodash

File: test/mocha.opts (hunk 3 above)
Comment (lumberj) [inline test/mocha.opts]: Removed

**Pull request merged**